ehcache-impl/src/test/java/org/ehcache/impl/internal/store/heap/bytesized/OnHeapStoreByteSizeTest.java

```diff
@@ -0,0 +1,97 @@
+package org.ehcache.impl.internal.store.heap.bytesized;
+
+import org.ehcache.Cache;
+import org.ehcache.CacheManager;
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.builders.CacheConfigurationBuilder;
+import org.ehcache.config.builders.CacheManagerBuilder;
+import org.ehcache.config.builders.ExpiryPolicyBuilder;
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.config.units.MemoryUnit;
+import org.ehcache.core.internal.statistics.DefaultStatisticsService;
+import org.ehcache.core.internal.statistics.DefaultTierStatistics;
+import org.junit.Test;
+
+import java.time.Duration;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+public class OnHeapStoreByteSizeTest {
```

Comment:
This test needs an @BeforeClass assumption to prevent it failing under Java 17. You can test this by running:
```
./gradlew -PtestVM=${JAVA_HOME_17} test
```